src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -129,6 +136,11 @@ public Person toModelType() throws IllegalValueException {
         }
         final Birthday modelBirthday = new Birthday(birthday);
 
+        if (!Goal.isValidGoal(goal)) {
+            throw new IllegalValueException(Goal.MESSAGE_CONSTRAINTS);
+        }
+        final Goal modelGoal = new Goal(Goal.ENUM_MAP.get(goal.toLowerCase(Locale.ROOT)));
```

Comment:
`Goal.fromGoalString(goal.toLowerCase(Locale.ROOT)` would be better. Can hide `ENUM_MAP`